src/parsing/effect_parsing.cpp

```diff
@@ -1164,12 +1171,42 @@ int32_t simplify_effect(uint16_t* source) {
 			}
 		} else {
 			auto sub_units_start = source + 2 + effect::effect_scope_data_payload(source[0]);
-
 			while(sub_units_start < source + source_size) {
-				auto const old_size = 1 + effect::get_generic_effect_payload_size(sub_units_start);
-				auto const new_size = simplify_effect(sub_units_start);
-
-				if(new_size != old_size) { // has been simplified, assumes that new size always <= old size
+				// [0      0     ] | clr_global_flag	[1      2     ] | clr_global_flag
+				// [0      1     ] | <flag>				[1      3     ] | <flag>
+				// --------------- | n = 1				...............
+				// [n      2n    ]
+				// [n      2n + 1]
+				int32_t old_size = 1 + effect::get_generic_effect_payload_size(sub_units_start);
+				int32_t new_size = simplify_effect(sub_units_start);
+				if((sub_units_start[0] & effect::code_mask) == effect::clr_global_flag) {
+					auto repeats = 0;
+					while(sub_units_start + old_size < source + source_size
+						&& (sub_units_start[old_size] & effect::code_mask) == effect::clr_global_flag
+						&& repeats < 8) {
+						old_size += 1 + effect::data_sizes[effect::clr_global_flag];
+						++repeats;
+					}
+					static const uint16_t fop_table[] = {
+						effect::clr_global_flag,
+						effect::fop_clr_global_flag_2,
+						effect::fop_clr_global_flag_3,
+						effect::fop_clr_global_flag_4,
+						effect::fop_clr_global_flag_5,
+						effect::fop_clr_global_flag_6,
+						effect::fop_clr_global_flag_7,
+						effect::fop_clr_global_flag_8
+					};
+					sub_units_start[0] = fop_table[repeats];
+					new_size = 1 + effect::data_sizes[fop_table[repeats]];
```

Comment:
There is a bug here. If there are other, non clear global flag effects after the sequence of global flag effects, this new size will chop them all off (you should add such an effect to one of your tests to make sure that it doesn't happen)